roles/backend_setup/tasks/vdo_create.yml

```diff
@@ -21,6 +14,40 @@
   loop: "{{ vdo_device_exists.results }}"
   when: item.stdout is defined and item.stdout == "0"
 
+- name: "set fact if it will at least install 1 vdo device"
+  set_fact:
+   gluster_infra_vdo_will_create_vdo: true
+  loop: "{{ gluster_infra_vdo }}"
+  loop_control:
+   index_var: index
+  when: item.device is defined and vdo_device_exists.results[index].stdout is defined and vdo_device_exists.results[index].stdout == "1"
+
+- name: Install VDO dependencies
+  #maybe use package module?
+  yum:
+   name: "{{ packages }}"
+  register: vdo_deps 
+  vars:
+    packages:
+      - kmod-kvdo
+      - vdo
+  when: ansible_os_family == 'RedHat' and gluster_infra_vdo_will_create_vdo is defined and gluster_infra_vdo_will_create_vdo==true
+        and (gluster_infra_installed_vdo_deps is not defined or not gluster_infra_installed_vdo_deps)
```

Comment:
gluster_infra_installed_vdo_deps is not defined at this point.
We are using it before it is defined by set_fact.

Reply:
That is correct, therefor it also checks if it's defined. This fact/variable is used to prevent running the yum install twice, in-case the vdo tasks are executed twice (e.g. phase #1 and phase #2). It's a minor optimisation, since yum should work on cached data, but in case it's not you don't have to wait for it a second time.